### PR TITLE
HTBHF-1459 Add EDD to paymentCycle.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

**Comment by mikeSmith08** [comment]
Can you change your commit comment please, its still referring to the ClaimStatus being changed to ACTIVE and hence nothing to do with this PR at all.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java [hunk 1]
```diff
@@ -48,11 +50,30 @@ void shouldCreateNewPaymentCycle() {
...
+        assertThat(result.getChildrenDob()).isEqualTo(datesOfBirth);
+        assertThat(result.getExpectedDeliveryDate()).isEqualTo(today);
+        assertThat(result.getTotalEntitlementAmountInPence()).isEqualTo(entitlement.getTotalVoucherValueInPence());
+        assertThat(result.getTotalVouchers()).isEqualTo(entitlement.getTotalVoucherEntitlement());
+    }
+
+    @Test
+    void shouldCreateNewPaymentCycleWithoutExpectedDueDateForEligibleClaimWithoutPregnancyVouchers() {
+        LocalDate today = LocalDate.now();
+        Claim claim = aClaimWithExpectedDeliveryDate(today);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java]: I think its misleading to have a valid EDD on the Claim for this test, we either need to not have one, or have one which is a couple of years in the past, that is a more representative scenario.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java [hunk 2]
```diff
@@ -7,12 +7,26 @@
...
-import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aValidVoucherEntitlement;
-import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aVoucherEntitlementWithEntitlementDate;
-import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aVoucherEntitlementWithZeroVouchers;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.*;
 
 public class PaymentCycleVoucherEntitlementTestDataFactory {
 
+    public static PaymentCycleVoucherEntitlement aPaymentCycleVoucherEntitlementWithPregnancyVouchers() {
+        return buildDefaultPaymentCycleVoucherEntitlement()
+                .voucherEntitlements(List.of(
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java]: Can we use `Collections.singletonList()` here, it's much clearer that it should only have a single element, `List.of` is great for anything else though.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleService.java [hunk 3]
```diff
@@ -59,14 +60,20 @@ public PaymentCycle createAndSavePaymentCycleForEligibleClaim(Claim claim,
                                                                   LocalDate cycleStartDate,
                                                                   PaymentCycleVoucherEntitlement voucherEntitlement,
                                                                   List<LocalDate> datesOfBirthOfChildren) {
+        LocalDate expectedDeliveryDate = voucherEntitlement.getVouchersForPregnancy() > 0
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleService.java]: Might it be worthwhile this being in a method called something like `getRelevantExpectedDeliveryDate` or `getExpectedDeliveryDateIfRelevant`?

Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleService.java]: Created getExpectedDeliveryDateIfRelevant

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java (hunk 2 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java]: Done.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java (hunk 1 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java]: Set to six months in the past.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/PaymentCycleVoucherEntitlementTestDataFactory.java]: Looks like you've done it below but not to this one???

**Comment by dwybourn** [comment]
> We also need to set the EDD in `DetermineEntitlementMessageProcessor.updateAndSavePaymentCycle`. Open to whether this should be done in this PR or another.

@YetAnotherMatt I can do in this PR. Just need to decide where the common logic should go.

**Comment by mikeSmith08** [comment]
The title of this PR has been changed, but I still think we need to correct the commit message, you might need to recreate the commit or branch to do this.

**Comment by YetAnotherMatt** [comment]
> The title of this PR has been changed, but I still think we need to correct the commit message, you might need to recreate the commit or branch to do this.

I missed the  fact that the commit message was wrong too!

**Comment by dwybourn** [comment]
> The title of this PR has been changed, but I still think we need to correct the commit message, you might need to recreate the commit or branch to do this.

@mikeSmith08 Rebased with correct commit message.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleServiceTest.java]: It's still showing today...

**Pull request closed without merging**